### ISSUE TITLE
docker-all example SSLHandshakeException: certificate_unknown

### ISSUE THREAD

**Issue by awoSYS** [eclipse-arrowhead/core-java-spring#281]
### Subject of the issue
When following the quickstart guide to run the docker-all example, I get the following error:

<details>
  <summary>
(Click to expand stack trace) avax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown 
</summary>

```
arrowhead_core          | 2021-02-08 11:44:59.058  INFO 012d83ff3da8 --- [           main] e.a.c.ApplicationInitListener            : Service Registry is unavailable at the moment, retrying in 15 seconds...
arrowhead_core          | 2021-02-08 11:45:14.183 ERROR 012d83ff3da8 --- [           main] e.a.c.h.HttpService                      : UnavailableServerException occurred at https://127.0.0.1:8443/serviceregistry/echo
arrowhead_core          | 2021-02-08 11:45:14.219 ERROR 012d83ff3da8 --- [           main] o.s.b.SpringApplication                  : Application run failed
arrowhead_core          | eu.arrowhead.common.exception.UnavailableServerException: Could not get any response from: https://127.0.0.1:8443/serviceregistry/echo
arrowhead_core          | 	at eu.arrowhead.common.http.HttpService.sendRequest(HttpService.java:169)
arrowhead_core          | 	at eu.arrowhead.common.http.HttpService.sendRequest(HttpService.java:186)
arrowhead_core          | 	at eu.arrowhead.common.ApplicationInitListener.checkServiceRegistryConnection(ApplicationInitListener.java:255)
arrowhead_core          | 	at eu.arrowhead.common.ApplicationInitListener.registerCoreSystemServicesToServiceRegistry(ApplicationInitListener.java:221)
arrowhead_core          | 	at eu.arrowhead.common.ApplicationInitListener.onApplicationEvent(ApplicationInitListener.java:103)
arrowhead_core          | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
arrowhead_core          | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
arrowhead_core          | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
arrowhead_core          | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
arrowhead_core          | 	at org.springframework.context.event.ApplicationListenerMethodAdapter.doInvoke(ApplicationListenerMethodAdapter.java:261)
arrowhead_core          | 	at org.springframework.context.event.ApplicationListenerMethodAdapter.processEvent(ApplicationListenerMethodAdapter.java:179)
arrowhead_core          | 	at org.springframework.context.event.ApplicationListenerMethodAdapter.onApplicationEvent(ApplicationListenerMethodAdapter.java:142)
arrowhead_core          | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
arrowhead_core          | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
arrowhead_core          | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
arrowhead_core          | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
arrowhead_core          | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
arrowhead_core          | 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
arrowhead_core          | 	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.finishRefresh(ServletWebServerApplicationContext.java:163)
arrowhead_core          | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
arrowhead_core          | 	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:142)
arrowhead_core          | 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:775)
arrowhead_core          | 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
arrowhead_core          | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:316)
arrowhead_core          | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
arrowhead_core          | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
arrowhead_core          | 	at eu.arrowhead.core.gateway.GatewayMain.main(GatewayMain.java:40)
arrowhead_core          | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
arrowhead_core          | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
arrowhead_core          | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
arrowhead_core          | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
arrowhead_core          | 	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
arrowhead_core          | 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
arrowhead_core          | 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
arrowhead_core          | 	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
arrowhead_core          | Caused by: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "https://127.0.0.1:8443/serviceregistry/echo": Received fatal alert: certificate_unknown; nested exception is javax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown
arrowhead_core          | 	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:744)
arrowhead_core          | 	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:710)
arrowhead_core          | 	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:598)
arrowhead_core          | 	at eu.arrowhead.common.http.HttpService.sendRequest(HttpService.java:158)
arrowhead_core          | 	... 34 more
arrowhead_core          | Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown
arrowhead_core          | 	at java.base/sun.security.ssl.Alert.createSSLException(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.Alert.createSSLException(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.TransportContext.fatal(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.Alert$AlertConsumer.consume(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(Unknown Source)
arrowhead_core          | 	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(Unknown Source)
arrowhead_core          | 	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:137)
arrowhead_core          | 	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:153)
arrowhead_core          | 	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:280)
arrowhead_core          | 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
arrowhead_core          | 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
arrowhead_core          | 	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
arrowhead_core          | 	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
arrowhead_core          | 	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:157)
arrowhead_core          | 	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
arrowhead_core          | 	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
arrowhead_core          | 	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
arrowhead_core          | 	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
arrowhead_core          | 	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
arrowhead_core          | 	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
arrowhead_core          | 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
arrowhead_core          | 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
arrowhead_core          | 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
arrowhead_core          | 	at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:87)
arrowhead_core          | 	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
arrowhead_core          | 	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:53)
arrowhead_core          | 	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:735)
arrowhead_core          | 	... 37 more
```
</details>

### Your environment
* OS: Ubuntu 18.04 LTS
* most recent version of eclipse-arrowhead/core-java-spring

### Steps to reproduce
```bash
# clone repo
git clone https://github.com/eclipse-arrowhead/core-java-spring.git

# prepare MySQL container (according to main repo README.md)
cd core-java-spring/docker-all/example/
docker volume create --name=mysql
cd ../..  # core-java-spring/
cp scripts/initSQL.sh docker-all/
cd docker-all
chmod +x initSQL.sh 
./initSQL.sh 

# pull latest base images
cd example/
docker-compose pull

# build jars
cd ../.. # core-java-spring/
mvn -DskipTests=true package -pl docker-all -am

# create another sql volume (according to docker-all/README.md, probably not required)
cd docker-all/example
docker volume create arrowhead_core_mysql

# start everything
docker-compose up --build
```

### Expected behaviour
AHFW core components start up without error.

### Actual behaviour
AHFW core components start up, service registry not reachable due to unknown_certificate error.

Any help appreciated. Thanks!

**Comment by pkrasnousov** [comment]
Have the same problem 
SQL log full of `2021-04-28T09:36:02.020136Z 76 [Note] Bad handshake`
SQL container is available and I can connect to it with credentials from the `properties` file

**Comment by tsvetlin** [comment]
In the meantime new docker images have been added. https://hub.docker.com/u/svetlint
4.3.0 images will be available later today.

**Comment by InRiPa** [comment]
Might be related. But I had the `bad handshake` as well.
For me the issue was, that on the remote server (Ubuntu 20.04) the default java version is
```
java --version
openjdk 11.0.11 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)
```

With this version, I get bad handshake errors as well. Upgrading to java 14 fixed this issue
```
java --version
openjdk 14.0.2 2020-07-14
OpenJDK Runtime Environment (build 14.0.2+12-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 14.0.2+12-Ubuntu-120.04, mixed mode, sharing)
```

Hope it helps.

**Comment by borditamas** [comment]
Hello,

we have just released the v4.6.1, where a more flexible docker approach has been taken and bug fixes were managed.
[Take a look](https://github.com/eclipse-arrowhead/core-java-spring/wiki/Docker) and please open a new issue if you still runs into some error.
